blueprints/hr_assistant/salary.py

```diff
@@ -0,0 +1,355 @@
+            else:
+                qa = qa.filter(field=field, gte=time_entities[0])
+
+    return qa
+
+
+def _apply_money_filter(qa, request, responder):
+    # Apply money filter given any numeric values, comparators or extreme conditions
+    # If
+    try:
+        qa, size = money_filter(qa, request, responder)
+        return qa, size
+    except Exception:
+        responder.reply("I see you are looking for the {function}, can you be more specific?")
+        responder.params.allowed_intents = ('general.get_aggregate',
+                                            'salary.get_salary_aggregate',
+                                            'date.get_date_range_aggregate',
+                                            'unsupported.unsupported', 'greeting.*')
+        responder.listen()
+        return
+
+
+def money_filter(qa, request, responder):
+    """
+    This function is used to filter any salary related queries, that may include a
+    comparator, such as, 'what percentage earns less than 20 dollars an hour? ' or an extreme,
+    such as, 'highest earning employee'.
+    """
+
+    num_entity = [float(e['value'][0]['value'])
+                  for e in request.entities
+                  if e['type'] in ('sys_number', 'sys_amount-of-money')]
+
+    try:
+        comparator_entity = [e for e in request.entities if e['type'] == 'comparator'][0]
+    except Exception:
+        comparator_entity = []
+
+    try:
+        extreme_entity = [e for e in request.entities if e['type'] == 'extreme'][0]
+    except Exception:
+        extreme_entity = []
+
+    # The money entity can have either be accompanied by a comparator, extreme or no entity.
+    # These are mutually exclusive of others and hence can only be queried separately from
+    # the knowledge base.
+
+    if comparator_entity:
+        comparator_canonical = comparator_entity['value'][0]['cname']
+
+        if comparator_canonical == 'more than' and len(num_entity) == 1:
+            gte_val = num_entity[0]
+            lte_val = 1000  # Default value since it is much above the hourly salary limit
+
+        elif comparator_canonical == 'less than' and len(num_entity) == 1:
+            lte_val = num_entity[0]
+            gte_val = 0
+
+        elif comparator_canonical == 'equals to':
+            gte_val = num_entity[0]
+            lte_val = num_entity[0]
+
+        elif len(num_entity) > 1:
+            gte_val = np.min(num_entity)
+            lte_val = np.max(num_entity)
+
+        # Apply filter iff numerical entity exists
+        try:
+            qa = qa.filter(field='money', gte=gte_val, lte=lte_val)
+        except Exception:
+            pass
+
+    elif extreme_entity:
+        qa, size = _resolve_extremes(request, responder, qa, extreme_entity, 'money', num_entity)
+
+    elif len(num_entity) >= 1:
+        qa = qa.filter(field='money', gte=np.min(num_entity), lte=np.max(num_entity))
+    size = 301
```

Comment:
Why is the size being reset to a magic number here?

Reply:
Resolved in an update. It's the overall size of the dataset. Set to a global variable now.